sos/plugins/openstack_placement.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (C) 2019 Red Hat, Inc., Lee Yarwood <lyarwood@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.plugins import Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin
+
+
+class OpenStackPlacement(Plugin):
+    """OpenStack Placement
+    """
+    plugin_name = "openstack_placement"
+    profiles = ('openstack', 'openstack_controller',)
+
+    var_puppet_gen = "/var/lib/config-data/puppet-generated/placement"
+
+    def setup(self):
+
+        # collect commands output only if the openstack-placement-api service
+        # is running
+        service_status = self.get_command_output(
+            "systemctl status openstack-placement-api.service"
+        )
+
+        in_container = self.running_in_container()
+
+        if (service_status['status'] == 0) or in_container:
+            placement_config = ""
+            # if containerized we need to pass the config to the cont.
+            if in_container:
+                placement_config = "--config-dir " + self.var_puppet_gen + \
+                                "/etc/placement/"
+            self.add_cmd_output(
+                "placement-manage " + placement_config + " db version",
+                suggest_filename="placement-manage_db_version"
+            )
+
+        if self.get_option("all_logs"):
+            self.add_copy_spec([
+                "/var/log/placement/",
+                "/var/log/containers/placement/",
+                "/var/log/containers/httpd/placement-api/"
+            ])
+        else:
+            self.add_copy_spec([
+                "/var/log/placement/*.log",
+                "/var/log/containers/placement/*.log",
+                "/var/log/containers/httpd/placement-api/*log",
+            ])
+
+        self.add_copy_spec([
+            "/etc/placement/",
+            self.var_puppet_gen + "/etc/placement/",
+            self.var_puppet_gen + "/etc/my.cnf.d/tripleo.cnf",
+            self.var_puppet_gen + "/etc/httpd/conf/",
+            self.var_puppet_gen + "/etc/httpd/conf.d/",
+            self.var_puppet_gen + "/etc/httpd/conf.modules.d/*.conf",
+        ])
+
+        if self.get_option("verify"):
+            self.add_cmd_output("rpm -V %s" % ' '.join(self.packages))
```

Comment:
No need for this - sos already makes this call if `--verify` is used. Also, this would fail when `DebianPlacement` is loaded.

Reply:
Good tip, removed. However the same thing is all over other OpenStack plugins.